neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,275 @@
+for generating the initial memory map (global sampling frequency), n_channels,
+n_samples.
+
+metadata['sapiens_base']['biointerface_map'] provides all the channel specific information
+like channel_names, channel_ids, channel_types.
+
+An additional note on channels. It appears that analog channels are called `pri` or
+`ai0` within the metadata where as digital channels are called `din0` or `dout0`.
+In this first implementation it is up to the user to do the appropriate channel slice
+to only get the data they want. 
+
+Zach McKenzie
+
+"""
+
+from __future__ import annotations
+from pathlib import Path
+import json
+
+import numpy as np
+
+from .baserawio import (
+    BaseRawIO,
+    _signal_channel_dtype,
+    _signal_stream_dtype,
+    _spike_channel_dtype,
+    _event_channel_dtype,
+)
+from neo.core import NeoReadWriteError
+
+
+class NeuronexusRawIO(BaseRawIO):
+
+    extensions = ["xdat", "json"]
+    rawmode = "one-file"
+
+    def __init__(self, filename: str | Path = ""):
+        """
+        The Allego Neuronexus reader for the `xdat` file format
+
+        Parameters
+        ----------
+        filename: str | Path, default: ''
+            The filename of the metadata file should end in .xdat.json
+
+        Notes
+        -----
+        * The format involves 3 files:
+            * The *.xdat.json metadata file
+            * The *_data.xdat binary file of all raw data
+            * The *_timestamps.xdat binary file of the timeestamp data
+        From the metadata the other two files are located within the same directory
+        and loaded.
+
+        * The metadata is stored as the metadata attribute for individuals hoping
+        to extract probe information, but neo itself does not load any of the probe
+        information
+
+        Examples
+        --------
+        >>> from neo.rawio import NeuronexusRawIO
+        >>> reader = NeuronexusRawIO(filename='abc.xdat.json')
+        >>> reader.parse_header()
+        >>> raw_chunk = reader.get_analogsignal_chunk(block_index=0
+                seg_index=0,
+                stream_index=0)
+        # this isn't necessary for this reader since data is stored as float uV, but
+        # this is included in case there is a future change to the format
+        >>> float_chunk = reader.rescale_signal_raw_to_float(raw_chunk, stream_index=0)
+
+        """
+
+        BaseRawIO.__init__(self)
+
+        if not Path(filename).is_file():
+            raise FileNotFoundError(f"The metadata file {filename} was not found")
+        if Path(filename).suffix != ".json":
+            raise NeoReadWriteError(
+                f"The json metadata file should be given, filename entered is {Path(filename).stem}"
+            )
+        meta_filename = Path(filename)
+        binary_file = meta_filename.parent / (meta_filename.stem.split(".")[0] + "_data.xdat")
+
+        if not binary_file.exists() and not binary_file.is_file():
```

Comment:
is_file also does the function of exists so there is some redundancy here.

Reply:
I think this might be residual from an old implementation that I switched. I think the original idea was someone might get a folder (which could exist) and so I added the check for being a file. But not sure anymore why I put both in. Haha.

Reply:
Just left for now.